### ISSUE TITLE
Exceeding line is not wrapped

### ISSUE THREAD

**Issue by lambdista** [olafurpg/scalafmt#238]
Original:
```scala
  private def a012345678901234567(b01234567890123456789: B01234567890123456789): Future[C012345678901[D012345678901]] = {
    ???
  }
```

Scalafmt:
```scala
  private def a012345678901234567(b01234567890123456789: B01234567890123456789): Future[C012345678901[D012345678901]] = {
    ???
  }
```

Actually I don't have a strong preference about how it should look like, provided that the line length doesn't exceed the max 120 characters I provided through the .scalafmt config file:
```
--maxColumn 120
```

Using
* 0.2.3 IntelliJ plugin.

**Comment by sjrd** [comment]
This probably happens because, even if wrapped, the line still exceeds the line limit. However IMO in that case scalafmt should try and minimize the length in excess, and produce:
```scala
  private def a012345678901234567(
      b01234567890123456789: B01234567890123456789): Future[C012345678901[D012345678901]] = {
    ???
  }
```
or, if the style permits it:
```scala
  private def a012345678901234567(
      b01234567890123456789: B01234567890123456789)
      : Future[C012345678901[D012345678901]] = {
    ???
  }
```

**Comment by olafurpg** [comment]
In the soon-to-be-released 0.2.4 I get the following output with --maxColumn 80:

```scala
// before
object a {
private def a012345678901234567(
      b01234567890123456789aaaaaaaaaaaaaaaaa: B01234567890123456789): Future[C012345678901[D012345678901]] = {
    ???
  }
}
// after
object a {
  private def a012345678901234567(
      b01234567890123456789aaaaaaaaaaaaaaaaa: B01234567890123456789)
    : Future[C012345678901[D012345678901]] = {
    ???
  }
}
```

**Comment by lambdista** [comment]
:+1: thank you.
